ipa-core/src/net/test.rs

```diff
@@ -140,63 +353,42 @@ impl TestConfigBuilder {
         self
     }
 
-    #[must_use]
-    pub fn build(self) -> TestConfig {
-        let mut sockets = None;
-        let ports = self.ports.unwrap_or_else(|| {
-            let socks = array::from_fn(|_| TcpListener::bind("localhost:0").unwrap());
-            let ports = socks
-                .each_ref()
-                .map(|sock| sock.local_addr().unwrap().port());
-            sockets = Some(socks);
-            ports
-        });
-        let (scheme, certs) = if self.disable_https {
-            ("http", [None, None, None])
+    /// Creates a HTTP1 or HTTP2 client config.
+    pub fn create_client_config(&self) -> ClientConfig {
+        self.use_http1
+            .then(ClientConfig::use_http1)
+            .unwrap_or_default()
+    }
+
+    /// Get all the MPC ports in a ring specified by the shard index.
+    fn get_ports_for_shard_index(&self, ix: ShardIndex) -> Vec<Option<u16>> {
```

Comment:
from the ergonomic perspective, a better type to return would be `Option<Vec<u16>>` here